src/NuGet.Services.AzureSearch/DependencyInjectionExtensions.cs

```diff
@@ -106,7 +106,24 @@ private static void RegisterAzureSearchStorageServices(ContainerBuilder containe
                 .Keyed<ICloudBlobClient>(key);
 
             containerBuilder
-                .Register<IVersionListDataClient>(c => new VersionListDataClient(
+                .Register<IVersionListDataClient>(c =>
+                {
+                    var options = c.ResolveOptional<IOptionsSnapshot<Db2AzureSearchConfiguration>>();
```

Comment:
Hm.... this seems hard to discover to me. I'm not a big fan of toggling business logic in DI.

Can we just not call the VersionListDataClient when `DisableVersionListWriters` is true?

Reply:
Done